aspnetcore/web-api/error-handling.md

```diff
@@ -0,0 +1,230 @@
+
+1. Create a well-known exception type named `HttpResponseException`:
+
+    ```csharp
+    public class HttpResponseException : Exception
+    {
+        public int Status { get; set; } = 500;
+    
+        public object Value { get; set; }
+    }
+    ```
+
+1. Create an action filter named `HttpResponseExceptionFilter`:
+
+    ```csharp
+    public class HttpResponseExceptionFilter : IActionFilter, IOrderedFilter
+    {
+        public int Order { get; set; } = int.MaxValue - 10;
+    
+        public void OnActionExecuting(ActionExecutingContext context) {}
+    
+        public void OnActionExecuted(ActionExecutedContext context)
+        {
+            if (context.Exception is HttpResponseException exception)
+            {
+                context.Result = new ObjectResult(exception.Value)
+                {
+                    Status = exception.Status,
+                };
+                context.ExceptionHandled = true;
+            }
+        }
+    }
+    ```
+
+1. In `Startup.ConfigureServices`, add the action filter to the filters collection:
+
+    ```csharp
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.AddControllers(options => 
+            options.Filters.Add(new HttpResponseExceptionFilter()));
+    }
+    ```
+
+## Validation failure error response
+
+For web API controllers, MVC responds with a `ValidationProblemDetails` response type when model validation fails. MVC uses the results of <xref:Microsoft.AspNetCore.Mvc.ApiBehaviorOptions.InvalidModelStateResponseFactory> to construct
+the error response for a validation failure. The following example uses the factory to change the default response type to `SerializableError`:
+
+::: moniker range=">= aspnetcore-3.0"
+
+[!code-csharp[](index/samples/3.x/Startup.cs?name=snippet_DisableProblemDetailsInvalidModelStateResponseFactory&highlight=4-13)]
+
+::: moniker-end
+
+::: moniker range="= aspnetcore-2.2"
+
+[!code-csharp[](index/samples/2.x/Startup.cs?name=snippet_DisableProblemDetailsInvalidModelStateResponseFactory&highlight=5-14)]
+
+::: moniker-end
+
+## Client error response
+
+An *error result* is defined as a result with an HTTP status code of 400 or higher. For web API controllers, MVC transforms an error result to a result with <xref:Microsoft.AspNetCore.Mvc.ProblemDetails>.
+
+::: moniker range=">= aspnetcore-3.0"
+
+The error response can be configured in one of the following ways:
+
+1. [Use ApiBehaviorOptions.ClientErrorMapping](#use-apibehavioroptionsclienterrormapping)
+1. [Implement ProblemDetailsFactory](#implement-problemdetailsfactory)
+
+::: moniker-end
+
+::: moniker range="<= aspnetcore-2.2"
+
+The error response can be configured as outlined in the [Use ApiBehaviorOptions.ClientErrorMapping](#use-apibehavioroptionsclienterrormapping) section.
+
+::: moniker-end
+
+### Use ApiBehaviorOptions.ClientErrorMapping
```

Comment:
Consider moving this section heading to inside the moniker range that ends on Line 189. That way it only shows up for >=3.0.

Then, remove that paragraph on Line 193 and the moniker range (Lines 191 and 195). The sentence on Line 199 can pick up with the coverage. (Drop Lines 191-195.)

It will all compose a bit better imo. ........... or not if ur too busy to mess with it. We can fix it up on a later PR for this topic.